smdebug/tensorflow/base_hook.py

```diff
@@ -126,60 +128,82 @@ def _get_worker_name(self) -> str:
         It is safe to return the CONFIG_DEFAULT_WORKER_NAME in this case.
         :return: str
         """
-        try:
+        assert self.distribution_strategy is not None
+        if self.distribution_strategy == TFDistributionStrategy.HOROVOD:
             import horovod.tensorflow as hvd
 
-            if hvd.size():
-                return f"worker_{hvd.rank()}"
-        except (ModuleNotFoundError, ValueError, ImportError):
-            pass
-
-        tf_config = os.getenv("TF_CONFIG")
-        if tf_config and is_parameter_server_strategy(tf_config):
-            return get_worker_id_from_tf_config(tf_config)
-        return CONFIG_DEFAULT_WORKER_NAME
+            return f"worker_{hvd.rank()}"
+        elif self.distribution_strategy == TFDistributionStrategy.MIRRORED:
+            # unused for this strategy
+            return DEFAULT_WORKER_NAME
+        elif self.distribution_strategy == TFDistributionStrategy.NONE:
+            return DEFAULT_WORKER_NAME
+        elif self.distribution_strategy == TFDistributionStrategy.UNSUPPORTED:
+            raise NotImplementedError
+        elif self.tf_config_json and is_parameter_server_strategy(self.tf_config_json):
+            return get_worker_id_from_tf_config(self.tf_config_json)
 
     def export_collections(self):
-        num_workers = self._get_num_workers()
         if self.save_all_workers is False:
             num_workers = 1
-            if (
-                self.distribution_strategy == TFDistributionStrategy.PARAMETER_SERVER_STRATEGY
-                or self.distribution_strategy == TFDistributionStrategy.HOROVOD
-            ):
+        else:
+            num_workers = self._get_num_workers()
+        self.collection_manager.set_num_workers(num_workers)
+        assert self.distribution_strategy is not None
+        if (
+            self.distribution_strategy == TFDistributionStrategy.PARAMETER_SERVER
+            or self.distribution_strategy == TFDistributionStrategy.HOROVOD
+        ):
+            if self.save_all_workers is False:
                 if self.worker != self.chief_worker:
                     return
-        self.collection_manager.set_num_workers(num_workers)
+        elif self.distribution_strategy == TFDistributionStrategy.MIRRORED:
+            if len(self.device_map):
+                for device, serialized_device in self.device_map.items():
+                    if self.save_all_workers is False and device != self.chief_worker:
+                        continue
+                    collection_file_name = f"{serialized_device}_collections.json"
+                    self.collection_manager.export(self.out_dir, collection_file_name)
```

Comment:
Nit style: I think it's a little cleaner to avoid 'continue' where possible, using instead:
```
if save_all_workers is True or device == self.chief_worker:
    collection_file_name = f"{serialized_device}_collections.json"
    self.collection_manager.export(self.out_dir, collection_file_name)
```

No strong feeling on this one though.

Reply:
Agreed, will change